lib/getDigest.js

```diff
@@ -0,0 +1,41 @@
+const crypto = require('crypto')
+
+const getDigest = (
+  {
+    headers: {
+      digest,
+      'content-type': contentType,
+      'content-encoding': contentEncoding
+    },
+    body
+  }, options
+) => {
+  const { formattedAlgorithm, digestAlgorithm, digestValue } = parseBodyDigestHeader(digest)
+
+  const calculatedDigest = calculateDigest({
+    algorithm: formattedAlgorithm,
+    message: JSON.stringify(body),
+    encoding: options.digestEncoding
+  })
+
+  if (calculatedDigest !== digestValue) {
+    throw new Error('Calculated Digest does not match Digest from Request Header')
+  }
+
+  return `${digestAlgorithm}=${calculatedDigest}`
+}
+
+const parseBodyDigestHeader = (requestDigestHeader) => {
```

Comment:
[nit] We could probably just handle this all inside the getDigest method. Pulling it out into its own method actually seems to add more code and this parse method wont ever be overridden or tested on its own.

I like the inclination to pull this out into its own method but in this case I think we aren't really gaining much by doing so. Up to you to remove if you want or not.

Reply:
Good point. There was more to this method before I refactored it. Doesn't make sense now to have it in a separate method. Changed.